carsus/io/cmfgen/base.py

```diff
@@ -657,8 +564,216 @@ def __init__(self, data, priority=10):
+                                                ).as_posix()
+
+                data[ion] = {}
+                lvl_parser = CMFGENEnergyLevelsParser(osc_fname)
+                lns_parser = CMFGENOscillatorStrengthsParser(osc_fname)
+                data[ion]['levels'] = lvl_parser.base
+                data[ion]['lines'] = lns_parser.base
+
+                if cross_sections:
+                    pho_flist = []
+                    try:
+                        for j, k in enumerate(ion_keys['pho']):
+                            pho_fname = BASE_PATH.joinpath(ion_keys['pho'][j]).as_posix()
+                            pho_flist.append(pho_fname)
+
+                    except KeyError:
+                        logger.warning(f'No `pho` data for {symbol} {ion[1]}.')
+
+                    data[ion]['phixs'] = []
+                    for l in pho_flist:
+                        pho_parser = CMFGENPhotoionizationCrossSectionParser(l)
+                        data[ion]['phixs'].append(pho_parser.base)
+
+                    if ion == (1,0):
+                        hyd_fname = BASE_PATH.joinpath('hyd_l_data.dat').as_posix()
+                        gbf_fname = BASE_PATH.joinpath('gbf_n_data.dat').as_posix()
+
+                        hyd_parser = CMFGENHydLParser(hyd_fname)
+                        gbf_parser = CMFGENHydGauntBfParser(gbf_fname)
+
+                        data[ion]['hyd'] = hyd_parser.base
+                        data[ion]['gbf'] = gbf_parser.base
+
+        return cls(data, priority)
+
+    @staticmethod
+    def cross_sections_squeeze(reader_phixs, ion_levels,
+                               hyd_phixs_energy_grid_ryd, hyd_phixs,
+                               hyd_gaunt_energy_grid_ryd, hyd_gaunt_factor):
+        """ Docstring """
+
+        phixs_table_list = []
+        n_targets = len(reader_phixs)
+
+        for i in range(n_targets):
+
+            target = reader_phixs[i]
+            lower_level_label = target.attrs['Configuration name']
+            cross_section_type = target.attrs['Type of cross-section']
+
+            # Remove the "[J]" term from J-splitted levels labels
+            ion_levels['label'] = ion_levels['label'].str.rstrip(']')
+            ion_levels['label'] = ion_levels['label'].str.split('[', expand=True)
+
+            try:
+                match = ion_levels.set_index('label').loc[[lower_level_label]]
+
+            except KeyError:
+                logger.warning(f'Level not found: \'{lower_level_label}\'.')
+                continue
+
+            lambda_angstrom = match['Lam(A)'].tolist()
+            level_number = (match['ID'] -1).tolist()
+
+            # Get statistical weights for J-splitted levels
+            match['w'] = match['g']/match.sum()['g']
+            w = match['w'].tolist()
+
+            # match is > 1 just for J-splitted levels
+            for j in range(len(match)):
+                threshold_energy_ryd = HC_IN_EV_ANGSTROM / lambda_angstrom[j] / RYD_TO_EV
```

Comment:
Why is this in angstrom? I find it quite difficult to follow these unit conversions.